File: app/backend/api/v1/endpoints/projects.py
from fastapi import APIRouter, Depends

from app.backend.schemas.project.response import ProjectRead, ProjectDescriptionRead
from app.backend.services.interfaces.project_service import IProjectService
from app.backend.api.deps import get_project_service

router = APIRouter()


@router.get("/projects", response_model=list[ProjectRead])
async def list_projects(service: IProjectService = Depends(get_project_service)):
    return await service.list_projects()


@router.get("/projects/{project_id}", response_model=ProjectRead)
async def get_project(
    project_id: str,
    service: IProjectService = Depends(get_project_service)
):
    return await service.get_project(project_id)


@router.get("/projects/{project_id}/description", response_model=ProjectDescriptionRead)
async def get_project_description(
    project_id: str,
    service: IProjectService = Depends(get_project_service)
):
    return await service.get_project_description(project_id)

File: app/frontend/src/api/projects.ts
import { api } from "@/api/client";
import type { ProjectDescriptionRead, ProjectRead } from "@/types/project";

export async function getProjects(): Promise<ProjectRead[]> {
  const res = await api.get<ProjectRead[]>("/projects");
  return res.data;
}

export async function getProject(id: string): Promise<ProjectRead> {
  const res = await api.get<ProjectRead>(`/projects/${encodeURIComponent(id)}`);
  return res.data;
}

export async function getProjectDescription(id: string): Promise<ProjectDescriptionRead> {
  const res = await api.get<ProjectDescriptionRead>(
    `/projects/${encodeURIComponent(id)}/description`
  );
  return res.data;
}
